include/deal.II/base/config.h.in

```diff
@@ -243,6 +243,15 @@ DEAL_II_NAMESPACE_CLOSE
 #  define DEAL_II_CXX20_REQUIRES(condition)
 #endif
 
+/**
+ * Wrapper macro around __builtin_expect().
+ */
+#ifdef DEAL_II_HAVE_BUILTIN_EXPECT
+#  define DEAL_II_BUILTIN_EXPECT(a, b) __builtin_expect((a), (b))
+#else
+#  define DEAL_II_BUILTIN_EXPECT(a, b) (a)
+#endif
+
```

Comment:
Why do you define the `DEAL_II_BUILTIN_EXPECT` macro here instead of `exceptio_macros.h`. It doesn't need to be visible outside that header file, right? In that case, we can `undef` it at the end of that file.